regression-test/suites/nereids_function_p0/scalar_function/IP.groovy

```diff
@@ -0,0 +1,101 @@
+    qt_sql_cidr_ipv6_all """ select id, ipv6_cidr_to_range(ip6, 16) from fn_test_ip_nullable order by id; """
+    qt_sql_cidr_ipv4_all """ select id, ipv4_cidr_to_range(ip4, 16) from fn_test_ip_nullable order by id; """
+
+
+    // test IPV4_STRING_TO_NUM/IPV6_STRING_TO_NUM (we have null value in ip4 and ip6 column in fn_test_ip_nullable table)
+    test {
+        sql 'select id, ipv6_string_to_num(ip6) from fn_test_ip_nullable order by id'
+        exception "Null Input"
+    }
+
+    test  {
+        sql 'select id, ipv4_string_to_num(ip4) from fn_test_ip_nullable order by id'
+        exception "Null Input"
+    }
+
+    // test ipv_num_to_string
+    qt_sql_num2string_ipv6 "select id, ipv6_num_to_string(ipv6_string_to_num_or_default(ip6)) from fn_test_ip_nullable order by id"
+    qt_sql_num2string_ipv4 "select id, ipv4_num_to_string(ipv4_string_to_num_or_default(ip4)) from fn_test_ip_nullable order by id"
+
+    // test INET_NTOA/INET6_NTOA
+    qt_sql_inet6_ntoa "select id, inet6_ntoa(ipv6_string_to_num_or_default(ip6)) from fn_test_ip_nullable order by id"
+    qt_sql_inet_ntoa "select id, inet_ntoa(ipv4_string_to_num_or_default(ip4)) from fn_test_ip_nullable order by id"
+
+    // test IPV4_STRING_TO_NUM_OR_DEFAULT/IPV6_STRING_TO_NUM_OR_DEFAULT
+    qt_sql_string2num_or_default_ipv6 "select id, ipv6_string_to_num_or_default(ip6) from fn_test_ip_nullable order by id"
+    qt_sql_string2num_or_default_ipv4 "select id, ipv4_string_to_num_or_default(ip4) from fn_test_ip_nullable order by id"
+
+    // test IPV4_STRING_TO_NUM_OR_NULL/IPV6_STRING_TO_NUM_OR_NULL
+    qt_sql_string2num_or_null_ipv6 "select id, ipv6_string_to_num_or_null(ip6) from fn_test_ip_nullable order by id"
+    qt_sql_string2num_or_null_ipv4 "select id, ipv4_string_to_num_or_null(ip4) from fn_test_ip_nullable order by id"
+
+    // test IS_IPV4_COMPAT/IS_IPV4_MAPPED
+    qt_sql_is_ipv4_compat "select id, is_ipv4_compat(ip6) from fn_test_ip_nullable order by id"
+    qt_sql_is_ipv4_mapped "select id, is_ipv4_mapped(ip6) from fn_test_ip_nullable order by id"
+
+    // test IS_IP_ADDRESS_IN_RANGE
+    def cidr_prefix_v6 = '2001:db8::/32'
+    def cidr_prefix_v4 = '::ffff:192.168.0.4/128'    
+    qt_sql_is_ip_address_in_range_ipv6 "select id, is_ip_address_in_range(ip6, '$cidr_prefix_v6') from fn_test_ip_nullable order by id"
+    qt_sql_is_ip_address_in_range_ipv4 "select id, is_ip_address_in_range(ip4, '$cidr_prefix_v4') from fn_test_ip_nullable order by id"
+    qt_sql_is_ip_address_in_range_null "select id, is_ip_address_in_range(ip6, null) from fn_test_ip_nullable order by id"
+    qt_sql_is_ip_address_in_range_null "select id, is_ip_address_in_range(ip4, null) from fn_test_ip_nullable order by id"
+
+    // test IS_IPV4_STRING/IS_IPV6_STRING
+    qt_sql_is_ipv4_string "select id, is_ipv4_string(ip4) from fn_test_ip_nullable order by id"
```

Comment:
This SQL implies `CAST(ip4 AS STRING)` since `is_ipv4_string/is_ipv6_string` need a string argument. You should add two columns ip4_str and ip6_str and use them for functions need string argument.

Reply:
done!